src/scenes/levelend-scene.ts

```diff
@@ -70,27 +69,31 @@ export class LevelEndScene {
     this.audioPlayer = new AudioPlayer();
     this.starCount = starCount;
     this.currentLevel = currentLevel;
-
+    // Subscribe to the LEVEL_END_BACKGROUND_TOGGLE event
+    gameStateService.subscribe(gameStateService.EVENTS.LEVEL_END_BACKGROUND_TOGGLE, this.toggleLevelEndBackground);
     this.isLastLevel =
       this.currentLevel !==
       this.data.levels[this.data.levels.length - 1].levelMeta.levelNumber &&
       this.starCount >= 2;
-
-    this.monster = new Monster(
-      this.canvas,
-      monsterPhaseNumber,
-      this.switchToReactionAnimation
-    );
+    // this.monster = new Monster(
+    //   this.canvas,
+    //   monsterPhaseNumber,
+    //   this.switchToReactionAnimation
+    // );
     this.showLevelEndScreen();  // Display the level end screen
     this.addEventListener();
     this.renderStarsHTML();
   }
-
-  showLevelEndScreen() {
-    // Make the levelEnd element visible by setting display to block
+  // Method to show/hide the Level End background
+  toggleLevelEndBackground = (shouldShow: boolean) => {
     if (this.levelEndElement) {
-      this.levelEndElement.style.display = 'block';
+      this.levelEndElement.style.display = shouldShow ? 'block' : 'none';
```

Comment:
Now that I can see the whole logic, I think it can be simplified.
My suggestion would be to;

Your method toggleLevelEndBackground will be retained.
```
toggleLevelEndBackground = (shouldShow: boolean) => {
    if (this.levelEndElement) {
      this.levelEndElement.style.display = shouldShow ? 'block' : 'none';
    }
  };
```
In the constructor since we need this to be set to block everytime levelend-scene loads.
```
constructor() {
this.toggleLevelEndBackground(true)
}
```
 In the level-end-scene I have a method handlePublishEvent that I used to reuse for publishing events to game state this is where you can hook up the removal.
```
private handlePublishEvent(shouldShowLoading: boolean, gamePlayData = null) {
    gamePlayData && gameStateService.publish(gameStateService.EVENTS.GAMEPLAY_DATA_EVENT, gamePlayData);
    gameStateService.publish(gameStateService.EVENTS.SCENE_LOADING_EVENT, shouldShowLoading);
    this.toggleLevelEndBackground(!shouldShowLoading) ---> your toggleLevelEndBackground for removal

  }
```
If you would check handlePublishEvent method will always be expecting a true for shouldShowLoading because this is always called whenever we are expecting a changed scene from levelend-scene to either gameplay or level selection so this is the best place to use or handle the css hiding.
With this you can pretty much disregard the game-state and scene-handler changes.

Reply:
oh wait, The reason to use the changes in scene-handler is cause there is a delay when hiding the background from level end screen, So I was making sure that it only hides when the screen redirects and not before that. let me share a recording with you @bernhardccodev

Reply:
https://github.com/user-attachments/assets/793ad0ef-1f62-4928-8ff9-0e19dba1edf9

@bernhardccodev for you reference. by adding the check in scenehandler i only hide the background when the screen is redirecting and not before that.